---
title: "A First Walk Through haPLOType"
author: "Thomas Ng and Eric C. Anderson"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

`haplot` comes pre-loadd with some wonderful, real example data to get you started learning its functionality.
This vignette will walk you through a look at the example data and introduce you to many of the features
of `haPLOType`.

The data that we will be exploring together are called `fish2_example.rds` and will appear in the Shiny app dropdown menu
after installation following the README on GitHub (reproduce that here, later).  These data are from a MiSeq run with 
about 100 amplicons sequenced on close to 300 individual fish.  The fish are all from a single population, but, in order
to demonstrate the "group" attribute, we put half of the fish into the group "pop1" and the other half of the fish into
the group "pop2".

## The Basics

The first thing you will want to get straight is that there are essentially four different "panels" 
on the haPLOType viewer that will always stay with you.  From top to bottom they are:

1. **The interaction tabsets:**  These are the three tabs that live at the top of the page with the light
blue background.  Any one of the three: ">field selection", "+read criteria", and "+locus annotation"
can be selected.  Whichever one is selected determines the content of the panel directly beneath---the
user input panel.
1. **The user input panel:** This panel has a light gray background.  If the ">field selection" tabset is
selected, then the user input panel will include dropdown boxes that let you choose a group, locus, or
an individual that you wish to focus on (this choice affects what gets shown in different plots and outputs,
etc.) It also includes "backwards-and-forwards" buttons, "<" and ">", that let you step individually through
loci (the buttons to the right of "Locus") or through individual (fish in this case---the buttons to the 
right of the "Indiv" dropdown box.)  These are very useful when you want to work your way through each 
locus individually and choose those that do (and do not) appear to be reliabble enough to include in a 
panel of amplicons.
1. **The main body panel:** This occupies most of haPLOType's screen real estate.  It is where the actual
plots, tables, and other results are rendered.  What gets rendered here depends on which main body panel
has been chosen.  That choice is made using the *main panel choice tabset* described below.
1. The **main panel choice tabset** is always down at the bottom, and is a tabset that allows you to choose the contents 
of the main body panel:
    a. *Panel:* which data set do you want to explore.  When you started haPLOType it should have had the 
    dropdown menu for "fish2_example.rds".
    b. *Summary Info: * let's you choose plot-summaries that give you a good overview of the characteristics of groups, individuals, or loci.
    c. *Filter Analysis:* let's you explore 1) a tabular summary (the "Broad Summary") of the number of haplotypes observed and 
    individuals called subject to different levels of filtering. 2) Distribution Plots that show the distribution of read depths
    and the allelic depth ratios of different haplotypes at individual loci, and 3) "Final Genotype" shows haplotype frequencies,
    and a visual summary of conformance to Hardy-Weinberg proportions of individual loci, and also portrays the position of the 
    variants within the sequence.
    d. *Inferential Analysis: * is a wrapper that integrates with the "Sr.MicroHap" Bayesian haplotype caller which is 
    in development still.
    e. *Output: * provides a place to get a number of tabular views of the data that are sortable and searchable.  This can
    be very useful when you want to dig into the nitty-gritty details of the short read data.  It also allows you to download
    filtered and called data and genotypes for later analysis.
    f. *About haPlot: * provides a little background about haplot and how to cite it and whom to contact about it.
    
With that background we are ready to start exploring the example data!

## Mission one: An overview of loci 

The first thing we are going to do is get an overview of the data with no filtering, focusing on 
different loci.  Not surprisingly, for this we choose *Summary Info --> By Locus*.  We also want to 
make sure that we are staring from the defaults in ">field selection" (ALL group, ALL loci, ALL individuals),
and under "+read criteria" you want to ensure that "Minimal Read Depth" and "Minimal Allelic Ratio" are
both set to 0.  (If you had changed these before then once you set them back to 0 you should hit the "apply"
button to apply the changes.)



Vignettes are long form documentation commonly included in packages. Because they are part of the distribution of the package, they need to be as compact as possible. The `html_vignette` output type provides a custom style sheet (and tweaks some options) to ensure that the resulting html is as small as possible. The `html_vignette` format:

- Never uses retina figures
- Has a smaller default figure size
- Uses a custom CSS stylesheet instead of the default Twitter Bootstrap style

## Vignette Info

Note the various macros within the `vignette` section of the metadata block above. These are required in order to instruct R how to build the vignette. Note that you should change the `title` field and the `\VignetteIndexEntry` to match the title of your vignette.

## Styles

The `html_vignette` template includes a basic CSS theme. To override this theme you can specify your own CSS in the document metadata as follows:

    output: 
      rmarkdown::html_vignette:
        css: mystyles.css

## Figures

The figure sizes have been customised so that you can easily put two images side-by-side. 

```{r, fig.show='hold'}
plot(1:10)
plot(10:1)
```

You can enable figure captions by `fig_caption: yes` in YAML:

    output:
      rmarkdown::html_vignette:
        fig_caption: yes

Then you can use the chunk option `fig.cap = "Your figure caption."` in **knitr**.

## More Examples

You can write math expressions, e.g. $Y = X\beta + \epsilon$, footnotes^[A footnote here.], and tables, e.g. using `knitr::kable()`.

```{r, echo=FALSE, results='asis'}
knitr::kable(head(mtcars, 10))
```

Also a quote using `>`:

> "He who gives up [code] safety for [code] speed deserves neither."
([via](https://twitter.com/hadleywickham/status/504368538874703872))
